Code/GraphMol/SmilesParse/CXSmilesOps.cpp

```diff
@@ -1352,8 +1353,34 @@ bool parse_enhanced_stereo(Iterator &first, Iterator last, RDKit::RWMol &mol,
     }
   }
   if (!atoms.empty()) {
+    // we need to do a bit of work to check whether or not we've already seen
+    // this particular StereoGroup (was Github #6050)
+    group_id = 10 * group_id + static_cast<unsigned int>(group_type);
+    std::vector<unsigned int> sgTracker;
+    mol.getPropIfPresent(cxsgTracker, sgTracker);
```

Comment:
Should sgTracker be part of the StereoGroup object?  This would remove the need for the implicit copy here.

Reply:
I don't think the IDs should be part of the stereogroups. They only important here when parsing and can then simply be added when writing the groups out (which may not be in the order in which they were read or stored).